### PR TITLE
absorb qol

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Absorb now takes 15 seconds instead of 45 seconds to do
Absorb now gives you ALL of the chatlogs of the person you absorbed, accessible in the evolution menu
Absorb now gives you blood instead of just hunger (you're shlurping the forbidden cranberry juice)
People who die by an absorb no longer deathgasp on death
Notes of the person you absorb are now viewable in your evolution menu

## Why It's Good For The Game
Did you know that absorb takes 45 seconds as of now? Yeah that's kinda fucked up. Other stuff makes metachecking harder and makes sense to just have.

## Images of changes
Screenshot 2023-11-25 132928
Screenshot 2023-11-25 133038

## Testing
Mmm yes waiter more skrells please!

## Changelog
:cl:
tweak: Absorb now takes 15 seconds instead of 45 seconds to do, gives you all of the targets chatlogs and notes (Viewable in the evo menu), gives you blood, and people who die from it no longer deathgasp on death
/:cl:

### CODE REVIEW

**Comment by Aquilar** [comment]
could you also make it give you the notes of the subject?

**Comment by GDNgit** [comment]
> could you also make it give you the notes of the subject?

does that already

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 1]
```diff
@@ -93,7 +100,9 @@
 	var/datum/action/changeling/evolution_menu/E = locate() in user.actions
 	SStgui.update_uis(E)
 
+	target.deathgasp_on_death = FALSE
 	target.death(FALSE)
+	target.deathgasp_on_death = TRUE
```
Comment (Contrabang) [inline code/modules/antagonists/changeling/powers/absorb.dm:105]:
```suggestion
	var/saved_var_or_something = target.deathgasp_on_death
	target.deathgasp_on_death = FALSE
	target.death(FALSE)
	target.deathgasp_on_death = saved_var_or_something
```

Comment (GDNgit) [reply code/modules/antagonists/changeling/powers/absorb.dm:105]: Actually I should just initial this

Comment (GDNgit) [inline code/modules/antagonists/changeling/powers/absorb.dm:105]:
```suggestion
	target.deathgasp_on_death = initial(target.deathgasp_on_death)
```

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 2]
```diff
@@ -93,7 +100,9 @@
 	var/datum/action/changeling/evolution_menu/E = locate() in user.actions
 	SStgui.update_uis(E)
 
+	target.deathgasp_on_death = FALSE
 	target.death(FALSE)
+	target.deathgasp_on_death = initial(target.deathgasp_on_death)
```
Comment (Contrabang) [inline code/modules/antagonists/changeling/powers/absorb.dm:105]: Nothing else changes this, but I am still somewhat skeptical of this. I personally think the method I suggested is better, but this is technically fine.

Comment (GDNgit) [reply code/modules/antagonists/changeling/powers/absorb.dm:105]: I think it's hella sus this is even a var I'll be real

**Comment by matttheficus** [comment]
As someone who's had funny log messages lag up my game when opening them in logging mode, I worry about performance on reading through the chatlogs.

Does this give you some fuckoff massive text string or does it get its own TGUI readable/searchable menu?

**Comment by GDNgit** [comment]
> As someone who's had funny log messages lag up my game when opening them in logging mode, I worry about performance on reading through the chatlogs.
> 
> Does this give you some fuckoff massive text string or does it get its own TGUI readable/searchable menu?

You only get the recent messages in your chat, the entire say log is added to your notes. There's also just a lot less stuff being sent vs with logging view

**Comment by matttheficus** [comment]
> You only get the recent messages in your chat, the entire say log is added to your notes.

I tend to actually use my in-game notes. Does this make your notes box a fuckoff massive mess of say logs? Breaking the usability of your character's Notes for say logs would be a bit gnarg, even if it is a QoL help for impersonating someone.

**Comment by GDNgit** [comment]
> I tend to actually use my in-game notes. Does this make your notes box a fuckoff massive mess of say logs? Breaking the usability of your character's Notes for say logs would be a bit gnarg, even if it is a QoL help for impersonating someone.


The notes from your victim's are directly below your own notes, this can make your notes long, but not really hard to navigate with CTRL-F always being an option and you always getting dropped off at the top. If you think that you get your say logs in your notes by default, you do not

> Could we also get an image showing the formatting in game of your notes screen after you succ someone with a decent sized say log?

Just long, does not fuck with the notes at the top

**Comment by SteelSlayer** [comment]
Has been a couple months since any commits have been pushed, though I did see you marked it as a draft recently. Are you still working on this?

Marking stale for now. PR is subject to close in a week.

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#define LING_ABSORB_RECENT_SPEECH			8	// The amount of recent spoken lines sent to the clings chat on absorbing a mob
 /datum/action/changeling/absorbDNA
```
Comment (Sirryan2002) [inline code/modules/antagonists/changeling/powers/absorb.dm:2]:
```suggestion
#define LING_ABSORB_RECENT_SPEECH			8	// The amount of recent spoken lines sent to the clings chat on absorbing a mob

/datum/action/changeling/absorbDNA
```
proper spacing

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 4]
```diff
@@ -41,7 +42,7 @@
 				to_chat(target, "<span class='danger'>You feel a sharp stabbing pain!</span>")
 				target.take_overall_damage(40)
 		SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("Absorb DNA", "[stage]"))
-		if(!do_mob(user, target, 150))
+		if(!do_mob(user, target, 5 SECONDS))
```
Comment (Sirryan2002) [inline code/modules/antagonists/changeling/powers/absorb.dm:45]: PR description says absorbtion is decreased from 45 seconds to 15 seconds but this only reduces it by 10 seconds, is this correct?

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 5]
```diff
@@ -56,27 +57,46 @@
...
+		for(var/datum/log_record/log as anything in GLOB.logging.get_logs_by_type(target.last_known_ckey, EMOTE_LOG))
+			target_say_log[2] += list(list("Chat" = log.what, "Time" = gameTimestamp(wtime = log.raw_time - 9.99)))
+		if(length(target.say_log) > LING_ABSORB_RECENT_SPEECH)
+			recent_speech = target.say_log.Copy(length(target.say_log) - LING_ABSORB_RECENT_SPEECH + 1, 0)
 		else
 			recent_speech = target.say_log.Copy()
 
+		var/account_pin
+		var/account_number
+		if(target.mind.initial_account)
```
Comment (Sirryan2002) [inline code/modules/antagonists/changeling/powers/absorb.dm:82]:
```suggestion
		if(istype(target.mind.initial_account))
```
type check preferred over null check here

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/antagonists/changeling/powers/absorb.dm (hunk 4 above)
Comment (GDNgit) [reply code/modules/antagonists/changeling/powers/absorb.dm:45]: yeah it's ran three times

File: code/modules/antagonists/changeling/powers/absorb.dm (hunk 5 above)
Comment (GDNgit) [reply code/modules/antagonists/changeling/powers/absorb.dm:83]: if someone gets a zero or another fucked up value in here we kinda want it to fail right?

Comment (Sirryan2002) [reply code/modules/antagonists/changeling/powers/absorb.dm:83]: I guess you could go either way, it's unlikely initial_account is gonna have a fucky value like that ever. Feel free to do whatever you think is best, a unit test would be better to catch any issues that would runtime here tbqh.

**Comment by S34NW** [comment]
Closing due to staleness. Please reopen or make a new PR when requested changes have been made.

**Pull request closed without merging**